backend/Dockerfile

```diff
@@ -1,12 +1,11 @@
 FROM node:alpine
 
-WORKDIR /usr/app
+WORKDIR /app
 
-COPY ./package*.json ./
+COPY . .
```

Comment:
note that this will disable caching - *every* code change will trigger a full `npm install`, whereas with your existing setup only changes to `package*.json` will trigger `npm install`

Reply:
Oh did not know that

Reply:
Doesn't this volume in docker-compose persist the node_modules so there won't be full `npm install`?
```
volumes:
  - /app/node_modules
```

Reply:
I'm pretty sure when docker runs the build, it hashes the content of each layer every time a `COPY` happens to determine what can be skipped - in this case, I *believe* the steps are:

* `compose` persists whatever
* rebuild is triggered
* `COPY . .` => `*.jsx` is changed for example => does not skip the next `RUN` with npm install, etc.

vs

* `compose` persists whatever
* rebuild is triggered
* `COPY ./package*.json ./` => contents are unchanged, skips the next `RUN` (does not run npm install)
* `COPY . /app` => `*.jsx` is changed for example, does not skip the next `RUN`

Reply:
compose volume mounting is intended for data, not build requirements

Reply:
Further reading: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache

> For the ADD and COPY instructions, the contents of the file(s) in the image are examined and a checksum is calculated for each file. The last-modified and last-accessed times of the file(s) are not considered in these checksums. During the cache lookup, the checksum is compared against the checksum in the existing images. If anything has changed in the file(s), such as the contents and metadata, then the cache is invalidated.